src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java

```diff
@@ -47,28 +50,110 @@
     private CuAttachmentServiceImpl attachmentService;
     private Attachment attachment;
     private Note noteVirus;
+    
+    private InputStream virusInputStream;
+    private InputStream goodInputStream;
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
     @Before
-    public void setUp() throws IOException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    public void setUp() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, IOException{
+        virusInputStream = setupInputStream(ATTACHMENT_TEST_FILE_PATH + File.separator + VIRUS_FILE_NAME);
+        goodInputStream = setupInputStream(ATTACHMENT_TEST_FILE_PATH + File.separator + GOOD_FILE_NAME);
+        noteVirus = setupMockNote(String.valueOf(new Guid()));
+        
         attachmentService = new CuAttachmentServiceImpl();
-        attachmentService.setKualiConfigurationService(new MockConfigurationService());
-        attachmentService.setAttachmentDao(new MockAttachmentDao());
-        attachmentService.setNoteService(new MockNoteService());
-        attachmentService.setAntiVirusService(new MockAntivirusService());
+        attachmentService.setKualiConfigurationService(buildMockConfigurationService());
+        attachmentService.setAttachmentDao(buildMockAttachmentDao());
+        attachmentService.setNoteService(buildMockNoteService());
+        attachmentService.setAntiVirusService(buildMockAntiVirusService());
 
         attachment = new Attachment();
         attachment.setObjectId(String.valueOf(new Guid()));
         attachment.setAttachmentIdentifier(attachment.getObjectId());
-        Note noteGood = setupMockNote();
+        Note noteGood = setupMockNote(String.valueOf(new Guid()));
         attachment.setNote(noteGood);
         createAttachmentFile(noteGood, GOOD_FILE_NAME);
 
-        noteVirus = setupMockNote();
+        
         createAttachmentFile(noteVirus, VIRUS_FILE_NAME);
     }
+    
+    @After
+    public void tearDown() throws IOException {
+        FileUtils.forceDelete(new File(TEST_PATH).getAbsoluteFile());
+        virusInputStream = null;
+        goodInputStream = null;
+        noteVirus = null;
+        attachmentService = null;
+    }
+    
+    private NoteService buildMockNoteService() {
+        NoteService mockNoteService = Mockito.mock(NoteService.class);
+        Mockito.when(mockNoteService.getNoteByNoteId(Mockito.any())).thenReturn(noteVirus);
```

Comment:
Are there any test cases where you would need to return the "good" note instead?

Reply:
the original code only returned noteVirus for the mock note service.  I think this meets the current testing needs.

Reply:
Ok, thanks for clarifying.